compiler/docgen.nim

```diff
@@ -1121,6 +1122,37 @@ proc genJsonItem(d: PDoc, n, nameNode: PNode, k: TSymKind): JsonItem =
           param["types"].add %($kind)
         result.json["signature"]["genericParams"].add param
 
+proc setDoctype(d: PDoc, n: PNode) =
+  ## Processes `{.doctype.}` pragma changing Markdown/RST parsing options.
+  if n != nil:
+    if n.len == 2:
+      var dt = ""
+      if n[1].kind == nkStrLit:
+        dt = n[1].strVal.toLowerAscii
+      elif n[1].kind == nkIdent:
+        dt = n[1].ident.s.toLowerAscii
+      else:
+        localError(d.conf, n.info, errUser,
+                       "unknown argument type $1 provided to doctype" % [
+                         $n[1].kind])
```

Comment:
```suggestion
      case n[1].kind
      of nkStrLit:
        dt = toLowerAscii(n[1].strVal)
      of nkIdent:
        dt = toLowerAscii(n[1].ident.s)
      else:
        localError(
          d.conf,
          n.info,
          errUser,
          "unknown argument type $1 provided to doctype" % [$n[1].kind]
        )
```
(that `localError` change can be skipped if that's what the style convention in the compiler code uses)